### PR TITLE
python312Packages.scales: run tests using pytestCheckHook

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/scales/default.nix [hunk 1]
```diff
@@ -16,11 +17,23 @@ buildPythonPackage rec {
     sha256 = "8b6930f7d4bf115192290b44c757af5e254e3fcfcb75ff9a51f5c96a404e2753";
   };
 
-  nativeCheckInputs = [ nose ];
+  patches = [
+    # Use html module in Python 3 and cgi module in Python 2
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scales/default.nix:21]:
```suggestion
    # Use html module in Python 3 and cgi module in Python 2
    # https://github.com/Cue/scales/pull/47
```

File: pkgs/development/python-modules/scales/default.nix [hunk 2]
```diff
@@ -16,11 +17,23 @@ buildPythonPackage rec {
...
+      hash = "sha256-xBlgkh1mf+3J7GtNI0zGb7Sum8UYbTpUmM12sxK/fSU=";
+    })
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
   propagatedBuildInputs = [ six ];
 
-  # No tests included
-  doCheck = false;
+  preCheck = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scales/default.nix:31]: This should be done in postPatch so that we notice early if it doesn't apply anymore

**Review by emilazy — COMMENTED** [review]
This is only used as an optional dependency in one other package, and hasn’t seen a release in almost a decade. Perhaps we should just drop it and remove the optional dependency from `cassandra-driver`?

If not, this should be migrated to `pyproject` at least.

**Comment by pyrox0** [comment]
I've made #348559 which incorporates the changes from this PR as well as a few extra changes that were noted above.

**Pull request closed without merging**